internal/provider/aws/provider.go

```diff
@@ -34,6 +39,24 @@ const (
 	TagKeyExternalIPName TagKey = TagKeyDomain + "/external-ip-name" // Tag key for external IP name
 )
 
+var (
+	// The duration for the items in the cache to expire (by default)
+	DefaultTTL = getEnvDuration("DEFAULT_TTL", 15*time.Minute)
+	// DefaultCleanupInterval triggers cache cleanup (lazy eviction) at this interval.
+	DefaultCleanupInterval = getEnvDuration("DEFAULT_CLEANUP_INTERVAL", time.Minute)
+)
+
+// getEnvDuration retrieves a duration value from an environment variable or returns a default value if not set
+// or an error occurs
+func getEnvDuration(envVar string, defaultValue time.Duration) time.Duration {
+	if value, exists := os.LookupEnv(envVar); exists {
+		if duration, err := time.ParseDuration(value); err == nil {
+			return duration
+		}
+	}
+	return defaultValue
+}
+
```

Comment:
You could use flags defined in the main function and give them to the provider constructor.